### PR TITLE
feat(exec): windows process runner

### PR DESCRIPTION
**Pull request by tilo-chen**
**Issue #, if available:**

**Description of changes:**
- Add vendor library for interacting with windows credential manager
- Add windows run as another user impl

**Why is this change necessary:**

**How was this change tested:**

**Any additional information or context required to review the change:**

**Checklist:**
 - [ ] Updated the README if applicable
 - [x] Updated or added new unit tests
 - [ ] Updated or added new integration tests
 - [ ] Updated or added new end-to-end tests
 - [ ] If your code makes a remote network call, it was tested with a proxy
 - [ ] You confirm that the change is backwards compatible

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java [hunk 1]
```diff
@@ -69,27 +67,33 @@ public Path which(String fn) {
 
     @Override
     public String[] getCommand() {
-        String[] decorated = cmds;
+        String[] decorated = Arrays.copyOf(cmds, cmds.length);
+        for (int i = 0; i < decorated.length; i++) {
+            final String arg = decorated[i];
+            // Space and \t require quoting otherwise will be split to more than one arg
```
Comment (tilo-chen) [inline src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java:73]: Equivalent to VERIFICATION_LEGACY: https://github.com/openjdk/jdk/blob/master/src/java.base/windows/classes/java/lang/ProcessImpl.java#L217-L226

File: src/main/java/com/aws/greengrass/util/platforms/windows/WindowsPlatform.java [hunk 2]
```diff
@@ -162,8 +162,9 @@ public void validateDefaultConfiguration(Map<String, Object> proposedDeviceConfi
 
             @Override
             public Optional<RunWith> generate(DeviceConfiguration deviceConfig, Topics config) {
-                // TODO set the actual user name
-                return Optional.of(RunWith.builder().user(System.getProperty("user.name")).build());
+                // TODO do proper checks like UnixRunWithGenerator
+                String user = Coerce.toString(config.find(RUN_WITH_NAMESPACE_TOPIC, POSIX_USER_KEY));
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/util/platforms/windows/WindowsPlatform.java:166]: we need a windows user field, right?

File: src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java [hunk 3]
```diff
@@ -69,27 +67,33 @@ public Path which(String fn) {
...
     @Override
-    protected Process createProcess() {
-        WindowsRunasProcess winProcess = new WindowsRunasProcess(null, userDecorator.getUser());
-        winProcess.setLpEnvironment(computeEnvironmentBlock());
-        winProcess.setLpCurrentDirectory(dir.getAbsolutePath());
-        winProcess.start(String.join(" ", getCommand()));
-        return winProcess;
+    protected Process createProcess() throws IOException {
+        if (needToSwitchUser()) {
+            WindowsRunasProcess winProcess = new WindowsRunasProcess(LOCAL_DOMAIN, userDecorator.getUser());
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java:87]: what do we do if we can't find the password for the user? Why are we forcing local domain? Why about networked accounts?

File: src/main/java/com/aws/greengrass/util/platforms/windows/WindowsPlatform.java (hunk 2 above)
Comment (tilo-chen) [reply src/main/java/com/aws/greengrass/util/platforms/windows/WindowsPlatform.java:166]: Yes. Can't really just posixUser it's confusing

File: src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java (hunk 3 above)
Comment (tilo-chen) [reply src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java:87]: Need to do verification in RunWithGenerator. Check like user exists and we have the credential

**Pull request closed without merging**